src/tunnel.cc

```diff
@@ -1357,8 +1388,13 @@ TunnelStateData::startConnecting()
 
     assert(!destinations->empty());
     assert(!transporting());
+
+    delete savedError; // may still be nil
+    savedError = nullptr;
+    request->hier.peer_reply_status = Http::scNone; // TODO: Move to startPeerClock()?
```

Comment:
We can address this TODO here, but it will affect FwdState code path (at least) as well. Perhaps it is best left for a dedicated PR.